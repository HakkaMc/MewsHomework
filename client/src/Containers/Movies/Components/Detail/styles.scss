.detail {
  position: sticky;
  top: 0;
}

.posterContainer {
  .poster {
    max-width: 100%;
  }
}
