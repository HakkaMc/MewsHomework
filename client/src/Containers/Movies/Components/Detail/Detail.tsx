import React from "react";
import { compose } from "recompose";
import { connect } from "react-redux";
import { Grid, Typography } from "@material-ui/core";

import { POSTER_URL } from "../../../../constants";
import { ReduxState } from "../../../../reducers";
import { Movie } from "../../../../reducers/Movies/model";
import styles from "./styles.scss";

export interface Props {
	selectedMovie?: Movie;
}

export const Detail: React.FC<Props> = ({ selectedMovie }: Props) => {
	if (!selectedMovie) return null;

	const imgUrl = `${POSTER_URL}${selectedMovie.poster_path}`;

	return (
		<Grid container spacing={2} className={styles.detail}>
			<Grid item className={styles.posterContainer} xs={12} md={6}>
				<img src={imgUrl} alt="No preview" className={styles.poster} />
			</Grid>
			<Grid item xs={12} md={6}>
				<Typography variant="h6">{selectedMovie.title}</Typography>
				<Typography variant="subtitle1">
					{selectedMovie.original_language}: {selectedMovie.original_title}
				</Typography>
				<Typography variant="subtitle1" noWrap>
					Average vote: {selectedMovie.vote_average} ({selectedMovie.vote_count}{" "}
					votes)
				</Typography>
			</Grid>
		</Grid>
	);
};

export default compose<Props, any>(
	connect(
		(state: ReduxState) => ({
			selectedMovie: state.movies.selectedMovie
		}),
		null
	)
)(Detail);
