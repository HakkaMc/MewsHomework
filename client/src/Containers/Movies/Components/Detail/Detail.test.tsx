import React from "react";
import { shallow } from "enzyme";

import { Detail, Props } from "./Detail";

describe("<Detail/>", () => {
	let props: Props;

	beforeEach(() => {
		props = {
			selectedMovie: undefined
		};
	});

	it("Match snapshot with selectedMovie=undefined", () => {
		props.selectedMovie = undefined;
		const wrapper = shallow(<Detail {...props} />);
		expect(wrapper.isEmptyRender()).toBe(true);
		expect(wrapper).toMatchSnapshot();
	});

	it("Match snapshot with mock selectedMovie", () => {
		props.selectedMovie = {
			id: 0,
			original_language: "en",
			original_title: "title",
			poster_path: "",
			title: "title",
			vote_average: 0,
			vote_count: 0
		};

		const wrapper = shallow(<Detail {...props} />);
		expect(wrapper.isEmptyRender()).toBe(false);
		expect(wrapper).toMatchSnapshot();
	});
});
