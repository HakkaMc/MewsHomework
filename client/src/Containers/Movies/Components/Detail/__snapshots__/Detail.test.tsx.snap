// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Detail/> Match snapshot with mock selectedMovie 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={2}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={6}
    xs={12}
  >
    <img
      alt="No preview"
      src="https://image.tmdb.org/t/p/w500"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={6}
    xs={12}
  >
    <WithStyles(ForwardRef(Typography))
      variant="h6"
    >
      title
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      variant="subtitle1"
    >
      en
      : 
      title
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      noWrap={true}
      variant="subtitle1"
    >
      Average vote: 
      0
       (
      0
       
      votes)
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`<Detail/> Match snapshot with selectedMovie=undefined 1`] = `""`;
